packages/twenty-front/src/modules/object-record/hooks/useFetchMoreRecordsWithPagination.ts

```diff
@@ -209,21 +210,30 @@ export const useFetchMoreRecordsWithPagination = <
 
   const totalCount = data?.[objectMetadataItem.namePlural]?.totalCount;
 
+  const recordConnection = data?.[objectMetadataItem.namePlural];
+
   const records = useMemo(
     () =>
-      data?.[objectMetadataItem.namePlural]
-        ? getRecordsFromRecordConnection<T>({
-            recordConnection: data?.[objectMetadataItem.namePlural],
-          })
-        : ([] as T[]),
+      getRecordsFromRecordConnection<T>({
```

Comment:
I understand the usage, but I think we should catch it earlier is it possible to have data?.[objectMetadataItem.namePlural] undefined? shouldn't we throw/return earlier in this case instead

Reply:
Since in React we try to avoid blocking initial renders where everything is loading and we should just allow the render flow to complete, I think it's preferable here to return [] since it will allow everything to show something and base itself on loading instead on trying to guess whether undefined means loading or error.

In this case there is already an error object that we can watch to handle errors, we shouldn't mix both.